Universal FE Randomizer/src/fedata/gba/general/GBAFEStatboost.java

```diff
@@ -0,0 +1,224 @@
+package fedata.gba.general;
+
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import fedata.gba.AbstractGBAData;
+import fedata.gba.GBAFEStatDto;
+
+/**
+ * Dataobject that contains the data of a GBAFE Statboost entry.
+ * 
+ * Each entry has 12 bytes: HP, POW, SKL, SPD, DEF, RES, LUK, MOVE, CON, ??, ??, ??. (?? = unused)
+ * 
+ */
+public class GBAFEStatboost extends AbstractGBAData {
+	
+	public GBAFEStatboostDao dao;
+	
+	public class GBAFEStatboostDao extends GBAFEStatDto {
+		public GBAFEStatboost parent;
+		public int mov;
+		public int con;
+		
+		public boolean statIndexIsHp(int index) {
+			return index == 0;
+		}
+		
+		public GBAFEStatboostDao(GBAFEStatboost parent, byte[] bytes) {
+			assert(bytes.length == 12); // HP, POW, SKL, SPD, DEF, RES, LCK, MOV, CON in that order followed by 3 useless bytes.
+			this.parent = parent;
+			
+			hp = asInt(bytes[0]);
+			str = asInt(bytes[1]);
+			skl = asInt(bytes[2]);
+			spd = asInt(bytes[3]);
+			def = asInt(bytes[4]);
+			res = asInt(bytes[5]);
+			lck = asInt(bytes[6]);
+			mov = asInt(bytes[7]);
+			con = asInt(bytes[8]);
+		}
+		
+		/**
+		 * Verifies that there is only one non-zero stat and throws an exception if there is more than one.
+		 * After verifying, returns the index of that stat.
+		 */
+		public int getIndexOfOnlyStat() {
+			List<Integer> stats = asList().stream().filter(i -> i != 0).collect(Collectors.toList());
+			if (stats.size() != 1) {
+				throw new UnsupportedOperationException("Tried finding the index of the only stat, but there is more than one.");
+			}
+			return asList().indexOf(stats.get(0));
+		}
+		
+		public void setStatAtIndex(int index, int stat) {
+			switch(index) {
+				case 0: hp = stat; break;
+				case 1: str = stat; break;
+				case 2: skl = stat; break;
+				case 3: spd = stat; break;
+				case 4: def = stat; break;
+				case 5: res = stat; break;
+				case 6: lck = stat; break;
+				case 7: mov = stat; break;
+				case 8: con = stat; break;
+			}
+		}
```

Comment:
Would it be possible to make these indices enums so they read a bit better? I think, later on, if you want to have special logic for some stats, it would be easier to reason about than coming back to this file to see which indices map to which stats. We kind of have it already with HP, which we've sidestepped with a dedicated `statIndexIsHP` method. 

As a potential implementation for a hypothetical `BoostedStat` enum, maybe `GBAFEStatboost` has a public static method that can generate 
 the `PoolDistributor<BoostedStat>` used in `StatboosterRandomizer`. The enum values can mirror the index, and it makes reasoning about the randomizer logic a bit more straightforward and readable.

Reply:
Yeah, that certainly makes sense, I'll see that I do that when I also resolve the conflicts that I see exist from the other merges